tests/db/test_bigchain_api.py

```diff
@@ -7,196 +7,68 @@
-        b.write_block(block1)
+        b.store_bulk_transactions([tx])
 
-        monkeypatch.setattr('time.time', lambda: 1000000020)
         transfer_tx = Transaction.transfer(tx.to_inputs(), [([alice.public_key], 1)],
                                            asset_id=tx.id)
         transfer_tx = transfer_tx.sign([alice.private_key])
-        block2 = b.create_block([transfer_tx])
-        b.write_block(block2)
+        b.store_bulk_transactions([transfer_tx])
 
-        monkeypatch.setattr('time.time', lambda: 1000000030)
         transfer_tx2 = Transaction.transfer(tx.to_inputs(), [([alice.public_key], 1)],
                                             asset_id=tx.id)
         transfer_tx2 = transfer_tx2.sign([alice.private_key])
-        block3 = b.create_block([transfer_tx2])
-        b.write_block(block3)
+        with pytest.raises(OperationError):
+            b.store_bulk_transactions([transfer_tx2])
 
-        # Vote both block2 and block3 valid
-        vote = b.vote(block2.id, b.get_last_voted_block().id, True)
-        b.write_vote(vote)
-        vote = b.vote(block3.id, b.get_last_voted_block().id, True)
-        b.write_vote(vote)
-
-        with pytest.raises(CriticalDoubleInclusion):
-            b.get_spent(tx.id, 0)
-
-    @pytest.mark.genesis
-    def test_get_spent_with_double_spend_detected(self, b, monkeypatch, alice):
-        from bigchaindb.exceptions import CriticalDoubleSpend
+    @pytest.mark.tendermint
+    def test_get_spent_with_double_spend_detected(self, b, alice):
         from bigchaindb.models import Transaction
+        from bigchaindb.common.exceptions import DoubleSpend
 
         tx = Transaction.create([alice.public_key], [([alice.public_key], 1)])
         tx = tx.sign([alice.private_key])
 
-        monkeypatch.setattr('time.time', lambda: 1000000000)
-        block1 = b.create_block([tx])
-        b.write_block(block1)
+        b.store_bulk_transactions([tx])
 
-        monkeypatch.setattr('time.time', lambda: 1000000020)
         transfer_tx = Transaction.transfer(tx.to_inputs(), [([alice.public_key], 1)],
                                            asset_id=tx.id)
         transfer_tx = transfer_tx.sign([alice.private_key])
-        block2 = b.create_block([transfer_tx])
-        b.write_block(block2)
-
-        monkeypatch.setattr('time.time', lambda: 1000000030)
         transfer_tx2 = Transaction.transfer(tx.to_inputs(), [([alice.public_key], 2)],
                                             asset_id=tx.id)
         transfer_tx2 = transfer_tx2.sign([alice.private_key])
-        block3 = b.create_block([transfer_tx2])
-        b.write_block(block3)
 
-        # Vote both block2 and block3 valid
-        vote = b.vote(block2.id, b.get_last_voted_block().id, True)
-        b.write_vote(vote)
-        vote = b.vote(block3.id, b.get_last_voted_block().id, True)
-        b.write_vote(vote)
+        with pytest.raises(DoubleSpend):
+            b.validate_transaction(transfer_tx2, [transfer_tx])
```

Comment:
Also judging by the previous state, we should test how `BigchainDB.get_spent` raises `DoubleSpend` when called with `current_transactions=[transfer_tx]`.

Reply:
What works for me is keeping everything as it is. I change all the block creation to a `store_bulk_transaction` and in the end when I call 
```
with pytest.raises(CriticalDoubleSpend):
     b.get_spent(tx.id, 0)
```
the `CriticalDoubleSpend` is raised and the test passes.

Reply:
I think later on in the file we test an actual DoubleSpend. With this it's doubled.

Reply:
But then this one will be doubled?